### PR TITLE
[observability][export-api] Write actor events

### PR DESCRIPTION
**Pull request by nikitavemuri**
## Why are these changes needed?
- Write actor events to file as part of the export API. This logic is only run if `RayConfig::instance().enable_export_api_write()` is `true`. Default value is `false`.
- Event write is called whenever a value in the actor event data schema is modified. Typically this occurs before writing ActorTableData to the GCS table or publishing the data for the dashboard

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_actor_manager.cc
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_actor_manager.cc:228]: very nit; but I prefer this style to avoid nested code;

```
if (!RayConfig::instance().enable_export_api_write()) {
    return
}

....
```

File: src/ray/gcs/gcs_server/gcs_server_main.cc
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_server_main.cc:93]: can you add comment why this source type is set?

**Review by rkooo567 — COMMENTED** [review]
Code looks good!

can you run name:many_actors and many_nodes_actor_test_on_v2 from release tests with the flag on?

**Comment by nikitavemuri** [comment]
> can you run name:many_actors and many_nodes_actor_test_on_v2 from release tests with the flag on?

Yeah, here's the test run https://buildkite.com/ray-project/release/builds/21701

**Comment by rkooo567** [comment]
@nikitavemuri did you run the test with the flag on?

**Review by rkooo567 — APPROVED** [review]
LGTM! Likely this can cause perf issue at large scale, and we will follow up perf after merging initial PRs

**Comment by can-anyscale** [comment]
Broke https://github.com/ray-project/ray/issues/47511 and is blocking release. I'm reverting to unblock.

**Pull request merged**
